pkg/reconciler/chart/provider_test.go

```diff
@@ -53,6 +53,27 @@ func TestProvider(t *testing.T) {
 		}
 	})
 
+	t.Run("Render filtered manifest", func(t *testing.T) {
+		// kyma components
+		ws, err := wsFactory.Get(kymaVersion)
+		require.NoError(t, err)
+
+		provider, err := NewDefaultProvider(wsFactory, log)
+		require.NoError(t, err)
+
+		providerWithFilter := provider.WithFilter(func(manifest string) (string, error) { return "", nil })
```

Comment:
You do not need `providerWithFilter`, when calling `WithFilter` the receiver is a pointer, you are adding the filter directly to `provider `.

The reason to return the provider in the "With" functions is the ability to do fluent programming (e.g. `x.WithA().WithB().WithC()`)